### ISSUE TITLE
Alternate setting/functionality for Pump tool

### ISSUE THREAD

**Issue by ladlon** [turtletooth/tahoma#223]
I'd love to see an alternate mode/setting for the Pump tool, where it doesn't add additional control points, but instead just affects the area between two control points (if you click on a line) or affects the thickness value at an existing control point (if you click on a control point).

In the case of clicking on the line, it would (I assume) set the thickness value for the two neighbouring points.

In the case of clicking on a control point, it would set the thickness for just that control point.

In both cases, it would ramp between the values of each of the points.

I'm not saying REPLACE the current behavior (as it obviously has it's merits), but have this as an alternate mode.  This would allow for keeping the point count unchanged, as well as serving for an easy way to affect a specific, existing line (rather than trying to guess the correct brush size for the tool).

Perhaps, select it via a modifier key (such as CTRL), as an additional alternative to a pulldown menu (for those who aren't using a keyboard to control things).

Maybe even have an additional modifier key which makes the Pump tool go into 'whole shape' mode, affecting the entire shape evenly (...even though there IS already a Thickness field in the toolbar... but this would allow you to qickly do a global adjustment while also making specific area tweeks, without having to switch off the Pump tool).

**Comment by artisteacher** [comment]
With the select tool, you can already evenly adjust the line thickness of a selection. You can even do so for the entire level, or the line thickness for a selected style across a whole level - or selected frames:
<img width="378" alt="Screen Shot 2020-09-25 at 7 11 20 PM" src="https://user-images.githubusercontent.com/21159116/94323267-1822cb00-ff63-11ea-9077-2b4569881781.png">
Choose the mode, select as necessary & then use the small pump at the bottom right corner:
<img width="94" alt="Screen Shot 2020-09-25 at 7 11 35 PM" src="https://user-images.githubusercontent.com/21159116/94323272-1eb14280-ff63-11ea-8f35-fd0d031f9d75.png">

**Comment by ladlon** [comment]
Certainly, but what I was suggesting is that by ALSO adding a modifier key 'adjust entire shape' feature to the pump tool, you could quickly/easily jump back and forth between adjusting specific sections, and making global overall adjustments to it, without having to switch between tools.
